Framework/Geometry/src/Crystal/IndexingUtils.cpp

```diff
@@ -1474,7 +1473,9 @@ double IndexingUtils::GetFirstMaxIndex(const double magnitude_fft[], size_t N,
   if (found_max) {
     double sum = 0;
     double w_sum = 0;
-    for (size_t j = i - 2; j <= i + 2; j++) {
+    // only include single extra index if we're too near the end
+    const size_t last_index = i < N - 2 ? i + 2 : i + 1;
+    for (size_t j = i - 2; j <= last_index; j++) {
```

Comment:
Is this easier to read when written as `for (size_t j = i - 2; j < std::min(N, i + 3) ; j++)`?

Reply:
That is nice, I like that. Thanks.